exonum/src/node/consensus.rs

```diff
@@ -626,7 +626,7 @@ impl NodeHandler {
             bail!("Received already processed transaction, hash {:?}", hash)
         }
 
-        if let Err(e) = self.blockchain.check_tx(&msg) {
+        if let Err(e) = Blockchain::check_tx(&self.blockchain.snapshot(), &msg) {
```

Comment:
Snapshot is already available here, we don't have to create another one. Or am I wrong?